src/viewers/viewer/Viewer.js

```diff
@@ -829,19 +844,16 @@ class Viewer extends OlObject {
 
     /**
      * Centers view on the middle of a geometry
+     * unless geometry is already in viewport,
```

Comment:
this is not what is currently happening cf. testing report

Reply:
That is describing the default behaviour of the method.
You can use the `forceCentre` argument to ensure that you get the shape centred.